### PR TITLE
New getVariable() API

### LINKED ISSUE #13752: job.getVariable
**Is your feature request related to a problem? Please describe.**
Whenever I'm using the Java client with variables, I can end up writing `job.getVariablesAsMap().get("name")` a lot. Of course, I can use `getVariablesAsType` with a record; but a shorthand method would be useful.

**Describe the solution you'd like**
`job.getVariable("name")`.

Under the hood, this makes a call to `job.getVariablesAsMap()`, and *caches* the map (to make it as performant as deserialising to a map, then accessing various variables in user code), then returns `.get("name")` from that map.

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

To reduce the boilerplate of doing `job.getVariablesAsMap().get("name")` we provide a new API `getVariable(name)`, allowing users to easily get a single variable by name.

## Related issues

closes #13752 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by korthout — CHANGES_REQUESTED** [review]
Great improvement 🚀 

❌ There's a small copy/paste mistake that must be adjusted

Please have a look 🙇

File: clients/java/src/main/java/io/camunda/zeebe/client/api/response/ActivatedJob.java
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/response/ActivatedJob.java:112]: true 🙇

**Comment by nicpuppa** [comment]
Thanks @korthout, please have another look 👀

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java
Comment (korthout) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:159]: On second thought, I see a potential issue with this. The JavaDoc does not specify what happens when no variable with that name exists. So, users might start to depend on the implementation, which is `null` or an NPE depending on the used map implementation (see [Map.get](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html#get(java.lang.Object)) documentation). 

🔧 It may be best to be explicit in our JavaDoc.

❌ We should also add a test case for the missing variable behavior. 

❓ This behavior may differ from a variable with `null` as value, so we must consider what behavior we like best here. What do you think is best?

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks @nicpuppa 🙇 

❌ On second thought, I noticed a problem we should resolve before merging. Please have a look

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:159]: Nice point 👍 

If a variable as `null` value we should just simply return it. But in case the user provide a key that is not present in the variables map probably we should raise an error. So the user is aware of that. Of course this should be well explained in the javadoc. What do you think ?

Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:159]: 👍 I like it

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java
Comment (koevskinikola) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:138]: :+1: Nice optimization!

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java
Comment (koevskinikola) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:152]: :thinking: I'm wondering if throwing an exception is a good approach here. Previously, a user could handle an unavailable variable by doing `Map#getOrDefault(...)`. Now, they would have to wrap their call in a try-catch block which would add several lines of code.

Would returning an `Optional.of(variable)` be nicer here?

Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:152]: I think that request a variable that is not available is an error, so in some way we should provide to the user that feedback. If we use `Optional` we don't differ from a variable that is available but with `null` as value.

Comment (koevskinikola) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java:152]: :+1: That's a good point. Additionally, this is a new method, so we wouldn't break any existing user code/expectations. All good from my side.

**Review by koevskinikola — APPROVED** [review]
LGTM :rocket:

**Comment by nicpuppa** [comment]
bors merge

**Review by korthout — APPROVED** [review]

**Comment by nicpuppa** [comment]
bors merge

**Pull request merged**
